src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.Utilities.cs

```diff
@@ -30,4 +31,25 @@ internal static string SanitizeMemberName(string memberName)
     private static Regex InvalidNameCharsRegex() => _invalidNameCharsRegex;
     private static readonly Regex _invalidNameCharsRegex = new("[^0-9A-Za-z_]", RegexOptions.Compiled);
 #endif
+
+    /// <summary>Invokes the MethodInfo with the specified target object and arguments.</summary>
+    private static object? ReflectionInvoke(MethodInfo method, object? target, object?[]? arguments)
+    {
+#if NET
+        return method.Invoke(target, BindingFlags.DoNotWrapExceptions, binder: null, arguments, culture: null);
+#else
+        try
+        {
+            return method.Invoke(target, BindingFlags.Default, binder: null, arguments, culture: null);
+        }
+        catch (TargetInvocationException e) when (e.InnerException is not null)
+        {
+            // If we're targeting .NET Framework, such that BindingFlags.DoNotWrapExceptions
+            // is ignored, the original exception will be wrapped in a TargetInvocationException.
+            // Unwrap it and throw that original exception, maintaining its stack information.
+            System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e.InnerException).Throw();
+            return null;
```

Comment:
Nit:
```suggestion
            throw;
```

Reply:
Isn't that redundant given that the previous line always throws?

Reply:
No more redundant than `return null;`, which will also never be reached. But using `throw` makes it easy to see at a glance that you never exit this catch block successfully.